algorithms/src/sorting/impl/Kokkos_SortImpl.hpp

```diff
@@ -195,18 +195,17 @@ void sort_onedpl(const Kokkos::Experimental::SYCL& space,
                 "SYCL execution space is not able to access the memory space "
                 "of the View argument!");
 
-  // Can't use Experimental::begin/end here since the oneDPL then assumes that
-  // the data is on the host.
-  static_assert(
-      ViewType::rank == 1 &&
-          (std::is_same<typename ViewType::array_layout, LayoutRight>::value ||
-           std::is_same<typename ViewType::array_layout, LayoutLeft>::value),
-      "SYCL sort only supports contiguous rank-1 Views.");
+  static_assert(ViewType::rank == 1, "SYCL sort only supports rank-1 Views");
+  if (view.stride(0) != 1) {
+    Kokkos::abort("SYCL sort only supports rank-1 Views with stride = 1".);
+  }
```

Comment:
This ignored LayoutTiled, and would compile with it

Reply:
ops, yes totally, i actually had it that way in the first place and then simplified too much. thanks for catching it !